worlds/witness/test/test_disable_non_randomized.py

```diff
@@ -0,0 +1,36 @@
+from ..rules import _has_lasers
+from ..test import WitnessTestBase
+
+
+class TestDisableNonRandomized(WitnessTestBase):
+    options = {
+        "disable_non_randomized_puzzles": True,
+        "shuffle_doors": "panels",
+        "early_symbol_item": False,
+    }
+
+    def test_unrandomized_locations_do_not_exist(self) -> None:
+        self.assert_location_does_not_exist("Orchard Apple Tree 5")
+
+    def test_locations_got_disabled_and_alternate_activation_triggers_work(self) -> None:
+        """
+        Test that specific Discarded Panels give extra lasers.
+        """
+
+        with self.subTest("Test that unrandomized locations are disabled."):
```

Comment:
FWIW, I hate this too. I don't hate it enough to say "Please rework the whole core test base system to avoid it", but I do hate it enough to write this comment which can be just marked as resolved after acknowledging my feelings.

Reply:
Yeah :/